### PR TITLE
feat: LPv2 message reorder

### PR DESCRIPTION
**Pull request by wischli**
# Description

* Updates messages to match v2 spec
* Update LP Integration tests based on `reorder` submodule branch

# Checklist:

- [x] I have added Rust doc comments to structs, enums, traits and functions
- [x] I have made corresponding changes to the documentation
- [x] I have performed a self-review of my code
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: pallets/liquidity-pools/src/message.rs [hunk 1]
```diff
@@ -1044,7 +994,7 @@ mod tests {
 				decimals: 15,
 				restriction_set: 1,
 			},
-			"040000000000000001811acd5b3f17c06841c7e41e9e04cb1b536f6d65204e616d65000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000053594d424f4c00000000000000000000000000000000000000000000000000000f01",
+			"0b0000000000000001811acd5b3f17c06841c7e41e9e04cb1b536f6d65204e616d65000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000053594d424f4c00000000000000000000000000000000000000000000000000000f01",
```
Comment (lemunozm) [inline pallets/liquidity-pools/src/message.rs:997]: NIT NIT suggestion:

Maybe this super obscure line becomes clear using the `concat!` macro: i.e:

```
concat!(
    "0b" //msg_id
    "0000000000000001", //pool_id
    "811acd5b3f17c06841c7e41e9e04cb1b536f6d65204e616d65", // tranche_id
    "00000000000000000000000000000000", //...
    ..
)
```

And becomes easier also to create new tests

File: pallets/liquidity-pools/src/message.rs [hunk 2]
```diff
@@ -314,17 +293,19 @@ pub enum Message {
...
 		tranche_tokens_payout: u128,
-		/// The remaining investment amount denominated in the `foreign` payment
-		/// currency. It reflects the sum of the unprocessed as well as the
-		/// processed but not yet collected amounts.
-		remaining_invest_amount: u128,
+		/// The fulfilled investment amount denominated in the `foreign` payment
+		/// currency. It reflects the amount of investments which were processed
+		/// independent of whether they were collected.
+		// TODO(@Luis): Apply delta instead of remaining to foreign investments
+		fulfilled_invest_amount: u128,
```
Comment (lemunozm) [inline pallets/liquidity-pools/src/message.rs:300]: In `FulfilledDepositRequest` message the fulfilled amount is not needed.

Anyway, I'm not sure if the solitary side has modified these fields yet. In that case, the error you're experiencing could be due to different format in these messages between us and solidity? (although sure you've already check that)

**Review by lemunozm — COMMENTED** [review]
I understand your concern about merging this with the pending TODOs in integration test. I would consider that merging in `feat/lpv2` at this moment does not represent final changes and we can have tech-debts, until having the base of lpv2 logic.

Could we add before merging this an `#[ignore]` before the failing tests? To allow green CI in the meantime for PRs on top of this.

Are the liquidity pools unitary tests are passing now?


Looks everything pretty good. Thanks for your efforts put on this 💯

File: runtime/integration-tests/src/cases/lp/investments.rs [hunk 3]
```diff
@@ -165,7 +155,7 @@ mod with_pool_currency {
 	use super::{utils, *};
 	use crate::cases::lp::utils as lp_utils;
 
-	#[test_runtimes(all)]
+	#[test_runtimes([development])]
```
[the remark this answers is not in the document]
Comment (wischli) [reply runtime/integration-tests/src/cases/lp/investments.rs:158]: Yes, exactly. Didn't wanna add a TODO for all these.

File: runtime/integration-tests/src/cases/liquidity_pools.rs [hunk 4]
```diff
@@ -679,6 +678,30 @@ mod utils {
...
+						pallet_liquidity_pools_gateway::Event::OutboundMessageSubmitted {
+							..
+						} => true,
+						_ => false,
+					}
+				} else {
+					false
+				}
+			})
+	}
```
[the remark this answers is not in the document]
Comment (wischli) [reply runtime/integration-tests/src/cases/liquidity_pools.rs:704]: I see your point. I have used this utility for ensuring, that an outbound message has not been sent for a particular closure which doesn't panic. So I'd like to keep it for now.

[the remark this answers is not in the document]
Comment (lemunozm) [reply runtime/integration-tests/src/cases/liquidity_pools.rs:704]: Sure! Indeed, I think, at this point, it's a good addition. Just wanted to leave my thought regarding this and the possible evolve of these tests

**Comment by lemunozm** [comment]
Do we now the right indexes to fix the UTs?

**Comment by wischli** [comment]
> Do we know the right indexes to fix the UTs?

Sorry, forgot to push that.

**Review by lemunozm — APPROVED** [review]

**Comment by lemunozm** [comment]
Once we rebase and fix the pending test, we should protect the `feat/lp-v2` branch to only merge if CI passes

**Pull request merged**